### PR TITLE
Avoid using the `Polytope(::AbstractVector)` constructor

### PR DESCRIPTION
**Pull request by eliascarv**
### Note
The `Polytope(::AbstractVector)` constructors are commented out to check that tests will pass without them.
### Decision
Should we remove these constructors?

### Update 2023-07-13 18:00
It has been decided that the `Polytope(::AbstractVector)` constructors are no longer needed and can be removed.

### CODE REVIEW

File: src/mesh.jl [hunk 1]
```diff
@@ -172,10 +172,12 @@ function vertices(g::Grid)
...
 
 function element(g::Grid, ind::Int)
-  inds = CartesianIndices(size(g) .+ 1)
   elem = element(topology(g), ind)
   type = pltype(elem)
-  type([cart2vert(g, inds[i]) for i in indices(elem)])
+  einds = indices(elem)
+  cinds = CartesianIndices(size(g) .+ 1)
+  v = ntuple(i -> cart2vert(g, cinds[einds[i]]), nvertices(type))
+  type(v)
```
Comment (juliohm) [inline src/mesh.jl]:
```suggestion
  verts = ntuple(i -> cart2vert(g, cinds[einds[i]]), nvertices(type))
  type(verts)
```

File: src/polytopes/hexahedron.jl [hunk 2]
```diff
@@ -39,5 +39,7 @@ function (h::Hexahedron)(u, v, w)
   )
 end
 
-Random.rand(rng::Random.AbstractRNG, ::Random.SamplerType{<:Hexahedron{Dim,T}}) where {Dim,T} =
-  Hexahedron(rand(rng, Point{Dim,T}, 8))
+function Random.rand(rng::Random.AbstractRNG, ::Random.SamplerType{Hexahedron{Dim,T}}) where {Dim,T}
+  v = ntuple(i -> rand(rng, Point{Dim,T}), 8)
+  Hexahedron(v)
```
Comment (juliohm) [inline src/polytopes/hexahedron.jl]: rename v --> verts

**Review by juliohm — CHANGES_REQUESTED** [review]
Thank you @eliascarv , it looks great. Attached minor review.

Can you also provide a comparison in terms of speed and memory allocation compared to the master branch? Do we gain something with these replacements?

**Comment by eliascarv** [comment]
## Benchmarks
Version info:

image

### Test

```
master:

Test Summary:       | Pass  Total     Time
Meshes.jl (Float32) | 4673   4673  2m31.4s

Test Summary:       | Pass  Total     Time
Meshes.jl (Float64) | 4678   4678  1m54.4s

PR:

Test Summary:       | Pass  Total     Time
Meshes.jl (Float32) | 4673   4673  2m29.7s

Test Summary:       | Pass  Total     Time
Meshes.jl (Float64) | 4678   4678  1m52.7s
```

### materialize
master:

Captura de tela de 2023-07-13 17-09-54

PR:

image

### discretize Dehn

master:

Captura de tela de 2023-07-13 17-23-01

PR:

Captura de tela de 2023-07-13 17-21-06

File: src/transforms.jl [hunk 3]
```diff
@@ -53,6 +53,14 @@ _reconstruct(points, ::G) where {G<:Geometry} = G(points)
 _reconstruct(points, domain::Domain) = _reconstruct(points, GeometrySet(domain))
 _reconstruct(points, mesh::Mesh) = SimpleMesh(points, topology(mesh))
 _reconstruct(points, ::PointSet) = PointSet(points)
+_reconstruct(points, ::PolyArea) = PolyArea(points)
+_reconstruct(points, ::Ring) = Ring(points)
+_reconstruct(points, ::Rope) = Rope(points)
+function _reconstruct(points, ::PL) where {PL<:Polytope}
+  N = nvertices(PL)
+  v = ntuple(i -> @inbounds(points[i]), N)
+  PL(v)
```
Comment (juliohm) [inline src/transforms.jl]: You can write this in one line as well, without the intermediate v variable.

**Review by juliohm — APPROVED** [review]
Awesome work @eliascarv. I believe this new internal representation in set of constructors will help us design the most performant algorithms at compile time.

**Pull request merged**